### PR TITLE
GeoJSON - Activate Strict Null Checks

### PR DESCRIPTION
**Pull request by denisname**
Activate Strict Null Checks in GeoJSON and made the necessary changes:
- Set strictNullChecks to true
- Make `geometry` and `properties` _optionally_ nullable
- Add tests

More explanations can be found in this comment: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23794#issuecomment-368652409 
See also: [GeoJson rfc7946 – Section 3.2 and 3.3](https://tools.ietf.org/html/rfc7946#section-3.2).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [GeoJson rfc7946 – Section 3.2 and 3.3](https://tools.ietf.org/html/rfc7946#section-3.2)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by tomwanzek** [comment]
@denisname Thanks for putting this PR in.

@Cobster @atd-schubert @JeffJacobson We were working on finally validating and activating `strictNullChecks` on the **d3-geo** definitions. In the process, we noticed the overly conservative `null`-ed union types, this PR rectifies. Thanks in advance for letting us know, if you would like to go ahead with the proposed changes here. We have a dependency on this PR being merged to pursue the changes to **d3-geo**.

**Comment by tomwanzek** [comment]
cc @Ledragon see above.

File: types/geojson/geojson-tests.ts [hunk 1]
```diff
@@ -231,7 +231,7 @@ interface TestProperty {
 }
 
 const testProps: TestProperty = {
-    foo: "bar",
+    foo: "baz",
```
Comment (atd-schubert) [inline types/geojson/geojson-tests.ts]: 😄

**Review by atd-schubert — APPROVED** [review]
Sorry for the late response...

**Comment by tomwanzek** [comment]
Thanks to everyone.

@armanio123 It appears that the merged change has not been published to `npm/@types` yet. Possibly hanging somewhere in a pipeline. Could you kindly give it a nudge? Thx in advance.

**Pull request merged**